File: app/src/main/java/br/com/flasco/leanworks/presentation/event/ui/EventDetailFragment.java
package br.com.flasco.leanworks.presentation.event.ui;

import androidx.databinding.library.baseAdapters.BR;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import br.com.flasco.leanworks.R;
import br.com.flasco.leanworks.databinding.EventDetailFragmentBinding;
import br.com.flasco.leanworks.domain.model.Event;
import br.com.flasco.leanworks.presentation.event.ui.dialog.CheckInDialogFragment;
import br.com.flasco.leanworks.utils.PresentationUtils;
import dagger.hilt.android.AndroidEntryPoint;


@AndroidEntryPoint
public class EventDetailFragment extends Fragment {

    private EventDetailViewModel mViewModel;
    private EventDetailFragmentBinding eventDetailFragmentBinding;
    private Callback callback;

    public static EventDetailFragment newInstance(String eventId) {
        EventDetailFragment eventDetailFragment = new EventDetailFragment();
        if(eventId != null && !eventId.isEmpty()) {
            Bundle args = new Bundle();
            args.putString("event_id", eventId);
            eventDetailFragment.setArguments(args);
        }
        return eventDetailFragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        mViewModel = new ViewModelProvider(this).get(EventDetailViewModel.class);
        eventDetailFragmentBinding = EventDetailFragmentBinding.inflate(inflater, container, false  );
        eventDetailFragmentBinding.setViewModel(mViewModel);
        eventDetailFragmentBinding.setLifecycleOwner(getViewLifecycleOwner());
        Bundle bundle = getArguments();
        if(bundle != null){
            int eventId = Integer.valueOf(bundle.getString("event_id"));
            mViewModel.setEventId(eventId);
        }
        subscribeUi();
        return eventDetailFragmentBinding.getRoot();
    }

    private void subscribeUi() {
        mViewModel.eventMutableLiveData.observe(getViewLifecycleOwner(), event -> {
//            eventDetailFragmentBinding.notifyPropertyChanged(BR.event);
        });
        eventDetailFragmentBinding.toolbar.setOnMenuItemClickListener(item -> {
            Log.d(TAG, "subscribeUi: "+item.getItemId());
            switch(item.getItemId()){
                case R.id.action_share:
                    Event event = mViewModel.eventMutableLiveData.getValue();
                    startActivity(PresentationUtils.shareIntent(event.getDescription()));
                    return true;
                case R.id.action_back:
                    getActivity().onBackPressed();
                    return true;
            }
            return false;
        });

        eventDetailFragmentBinding.fab.setOnClickListener(v -> {
            CheckInDialogFragment checkInDialogFragment = new CheckInDialogFragment();
            Bundle bundle = new Bundle();
            bundle.putString("event_id", mViewModel.eventMutableLiveData.getValue().getId());
            checkInDialogFragment.setArguments(bundle);
            checkInDialogFragment.show(getActivity().getSupportFragmentManager(), "checkin_dialog");
        });
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
    }

    private static final String TAG = "EventDetailFragment";

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        Log.d(TAG, "onOptionsItemSelected: "+item.getItemId());
        return super.onOptionsItemSelected(item);
    }

    public interface Callback {
        void checkIn(Event event);
    }
}

File: app/src/main/java/br/com/diogomoreno/events/presentation/event/ui/EventDetailFragment.kt
package br.com.diogomoreno.events.presentation.event.ui

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import br.com.diogomoreno.events.domain.model.Event
import br.com.diogomoreno.events.presentation.event.ui.dialog.CheckInDialogFragment
import br.com.diogomoreno.events.utils.PresentationUtils
import br.com.flasco.leanworks.R
import br.com.flasco.leanworks.databinding.EventDetailFragmentBinding
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class EventDetailFragment : Fragment() {
    private var mViewModel: EventDetailViewModel? = null
    private var eventDetailFragmentBinding: EventDetailFragmentBinding? = null
    private val callback: Callback? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        mViewModel = ViewModelProvider(this).get(EventDetailViewModel::class.java)
        eventDetailFragmentBinding = EventDetailFragmentBinding.inflate(inflater, container, false)
        eventDetailFragmentBinding!!.viewModel = mViewModel
        eventDetailFragmentBinding!!.lifecycleOwner = viewLifecycleOwner
        val bundle = arguments
        if (bundle != null) {
            val eventId = Integer.valueOf(bundle.getString("event_id")!!)
            mViewModel!!.setEventId(eventId)
        }
        subscribeUi()
        return eventDetailFragmentBinding!!.root
    }

    private fun subscribeUi() {
        mViewModel!!.eventMutableLiveData.observe(viewLifecycleOwner, Observer { event: Event? -> })
        eventDetailFragmentBinding!!.toolbar.setOnMenuItemClickListener { item: MenuItem ->
            Log.d(TAG, "subscribeUi: " + item.itemId)
            when (item.itemId) {
                R.id.action_share -> {
                    val event = mViewModel!!.eventMutableLiveData.value
                    startActivity(PresentationUtils.shareIntent(event?.description))
                    return@setOnMenuItemClickListener true
                }
                R.id.action_back -> {
                    activity!!.onBackPressed()
                    return@setOnMenuItemClickListener true
                }
            }
            false
        }
        eventDetailFragmentBinding!!.fab.setOnClickListener { v: View? ->
            val checkInDialogFragment = CheckInDialogFragment()
            val bundle = Bundle()
            bundle.putString("event_id", mViewModel!!.eventMutableLiveData.value?.id)
            checkInDialogFragment.arguments = bundle
            checkInDialogFragment.show(activity!!.supportFragmentManager, "checkin_dialog")
        }
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        Log.d(TAG, "onOptionsItemSelected: " + item.itemId)
        return super.onOptionsItemSelected(item)
    }

    interface Callback {
        fun checkIn(event: Event?)
    }

    companion object {
        fun newInstance(eventId: String?): EventDetailFragment {
            val eventDetailFragment = EventDetailFragment()
            if (eventId != null && !eventId.isEmpty()) {
                val args = Bundle()
                args.putString("event_id", eventId)
                eventDetailFragment.arguments = args
            }
            return eventDetailFragment
        }

        private const val TAG = "EventDetailFragment"
    }
}